source/yats/include/yats/queue.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <mutex>
+#include <queue>
+
+namespace yats
+{
+
+template <typename ValueType>
+class queue
+{
+    using lock = std::lock_guard<std::mutex>;
+public:
+    using value_type = ValueType;
+
+    ValueType extract()
```

Comment:
Why not rename extract to pop?

Reply:
pop has a different interface

Reply:
That is true... Then I don't like the C++ interface. For me, pop should return and remove the element. But ok.